src/main/java/org/spongepowered/asm/mixin/injection/struct/MemberInfo.java

```diff
@@ -444,7 +444,8 @@ public MemberInfo validate() throws InvalidMemberDescriptorException {
                 throw new InvalidMemberDescriptorException("Invalid owner: " + this.owner);
             }
             try {
-                if (!this.owner.equals(Type.getType(this.owner).getDescriptor())) {
+                String ownerType = "L" + this.owner + ";";
+                if (!ownerType.equals(Type.getType(ownerType).getDescriptor())) {
```

Comment:
Although this will fix an owner type check it won't make it less stupider. 
Literally what we do now:
1. We write all chars between first "L" and ";" in string 'owner'. 'owner' now contains no "L" or ";" chars in MemberInfo parse() method.
2. We attaching "L" to beginning of our string and ";" to end.
3. Since we always attach "L" to beginning we always will get a type with 'sort' variable equal to 10 in Type getType() method.
4.When we getting descriptor in Type instance we gain at stage 3 for 'sort==10' we get string with "L" attached to beginning of our string and ";" to end.
Therefore this check is literally: ("L"+somestring+";").equals("L"+somestring+";")

Reply:
That check is exactly what we want. If `owner` is an improper Type string then it'll error when converting.

Reply:
I can write "L666;" and it will pass this check.

Reply:
Everything incorrect is catched 3 lines above by
```
!this.desc.matches("^(\\([\\w\\p{Sc}\\[/;]*\\))?\\[*[\\w\\p{Sc}/;]+$")
```